README.md

```diff
@@ -33,14 +32,29 @@ require 'surro-gate'
 proxy = SurroGate.new
 
 # Create two TCP socket connections
-left = TCPSocket.new('localhost', 3333)
-right = TCPSocket.new('localhost', 2222)
+sock_1 = TCPSocket.new('localhost', 1111)
+sock_2 = TCPSocket.new('localhost', 2222)
+sock_3 = TCPSocket.new('localhost', 3333)
+sock_4 = TCPSocket.new('localhost', 4444)
 
 # Push the sockets to the proxy
-proxy.push(left, right)
-
-# Wait for the communication to finish
-proxy.wait
+proxy.push(sock_1, sock_2)
+proxy.push(sock_3, sock_4)
+
+loop do
+  # Call the selector with a timeout value in milliseconds
+  proxy.select(1000)
+
+  # Do some hard work
+  proxy.each_ready do |left, right|
+    begin
+      right.write_nonblock(left.read_nonblock(4096))
```

Comment:
doesn't this return the count of characters written?

Reply:
it does

Reply:
lol. And aren't you curious if all characters went through?